src/dune_rules/workspace.ml

```diff
@@ -6,10 +6,39 @@ open Dune_lang.Decoder
    simplicity *)
 let syntax = Stanza.syntax
 
+let all_binaries e =
+  let open Dune_env.Stanza in
```

Comment:
nitpick: we do not use local opens to qualify fields in the dune code base. Instead, we prefer annotating the values with the type of the record.